### PR TITLE
video viewer

### LINKED ISSUE #3688: Meeting - Create the video viewer
Create a video player without controls that react to parameters in the data model
JWPlayer can probably be used to achieve that

link to epic #3679

### PR DESCRIPTION
**Pull request by pldespaigne**
fix #3688

### CODE REVIEW

File: apps/festival/festival/src/app/marketplace/event/session/session.component.html [hunk 1]
```diff
@@ -44,7 +44,7 @@ <h3 *ngIf="event.isOwner">You are the owner of this meeting</h3>
           <div [fxFlex]="mediaContainerSize">
             <media-viewer [event]="event"></media-viewer>
           </div>
-          <div [fxFlex]="visioContainerSize">VIDEO CONFERENCE</div>
+          <div [fxFlex]="visioContainerSize" class=visio>VIDEO CONFERENCE</div>
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/marketplace/event/session/session.component.html:47]: add "" around your class.

File: libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+@Component({
+  selector: '[ref] [control] event-video-viewer',
+  templateUrl: './video-viewer.component.html',
+  styleUrls: ['./video-viewer.component.scss'],
+  encapsulation: ViewEncapsulation.None, // We use `None` because we need to override the nested jwplayer css
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VideoViewerComponent implements AfterViewInit {
+
+  private player: any;
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts:22]: What about creating a type for that ?

File: libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    @Inject(DOCUMENT) private document: Document,
+    private eventQuery: EventQuery,
+    private authQuery: AuthQuery,
+    private mediaService: MediaService,
+    private functions: AngularFireFunctions,
+  ) { }
+
+  async initPlayer() {
+
+    const { id } = this.eventQuery.getActive();
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts:55]: I don't think it's a good idea to bind a reusable component like that to akita's query. What about adding this information in the input ?

File: libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    const privateVideo = this.functions.httpsCallable('privateVideo');
+    const { error, result } = await privateVideo({ eventId: id, ref: this.ref }).toPromise();
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts:58]: What if it's not an event video ? Like the trailer of a movie in the movie page ?

File: libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    if (!!error) {
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts:60]: Would be nice to have error codes to display the right message if needed.

**Review by GrandSchtroumpf — APPROVED** [review]

File: libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts (hunk 4 above)
Comment (pldespaigne) [reply libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts:58]: the functions will handle it and will return the corresponding error

File: libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts (hunk 2 above)
Comment (pldespaigne) [reply libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts:22]: There is a lot of thing to put in this type (https://developer.jwplayer.com/jwplayer/docs/jw8-javascript-api-reference).
Best would be to do it in a separate PR to the @types repo, but I'm not sure to have the time to do it ^^

File: libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts (hunk 5 above)
Comment (pldespaigne) [reply libs/media/src/lib/components/viewers/video-viewer/video-viewer.component.ts:60]: I've opened a dedicated issue for that #3926

**Pull request merged**